companion/index.js

```diff
@@ -54,14 +55,26 @@ settingsStorage.onchange = function(evt) {
         sendData({key: "clear"});
         JSON.parse(evt.newValue).forEach(element => {
             fetchEntity(address(), Token, element["name"]);
-        })
+        });
     }
     else if (evt.key === "force") {
         let data = JSON.parse(evt.newValue);
         sendData({key: "force", value: data});
     }
 }
 
+// Settings changed while companion was not running
+if (companion.launchReasons.settingsChanged) {
+    const keys = ["url", "port", "token", "force"];
+    keys.forEach(function(keyName, index, array) {
+        sendData({key: keyName, value: settingsStorage.getItem(keyName)});
+    });
+    sendData({key: "clear"});
+    settingsStorage.getItem("entities").forEach(element => {
```

Comment:
```
[17:57:56] Companion: TypeError: Cannot read properties of null (reading 'forEach')
  ? at companion/index.js:73,21
[17:57:56] Companion: Uncaught TypeError: Cannot read properties of null (reading 'forEach')                                                                                                     (companion/index.js:73,5)
```
this warning happens after every time i enter a value in the settings bit with the app closed on my watch

Reply:
also after setting an entities i get this error
```
[18:01:56] Companion: TypeError: o.settingsStorage.getItem(...).forEach is not a function
  ? at companion/index.js:73,21
[18:01:56] Companion: Uncaught TypeError: o.settingsStorage.getItem(...).forEach is not a function                                                                                               (companion/index.js:73,5)
```

Reply:
Thanks for testing.
Thought I can fix it fast, but apparently it needs a bit more testing. My JS skills are not good enough :)... Sorry, but then it takes a bit longer, at least till I got my HA up and running again.

Reply:
no rush @smirko-dev! im having a look myself but im the same at times haha

Reply:
getItem appears to return a string, so the "forEach" is not a thing. Something like this appears to solve the issue here:
```
  JSON.parse(settingsStorage?.getItem("entities"))?.forEach((element) => {
    fetchEntity(address(), Token, element["name"]);
  });
```